---
layout:     post
title:      Life Sciences
subtitle:   Transposable elements
date:       2019-04-08
author:     Yingshan Li
header-img: img/sunrise.jpg
catalog: true
tags:
    - Transposable elements
---


#### 1. Nine aacts about transposons


1. Transposons come in many different forms and shapes
2. Transposons are not randomly distributed in the genome: guided by opposite selective forces, a balancing act of facilitating future propagation while mitigating deleterious effects on host cell function.
3. Transposons are extensive source of mutations and genetic polymorphisms: most TE insertion sites are specific to species which is absent at orthologous sites of even closest relatives. TE insertions rarely provide immediate fitness advantages, and those reaching fixation in population do so largely by genetic drift and are subsequently eroded by point mutations that accumulate neutrally resulting in transposons that are no longer transposable.
4. Transposons are associated with genome rearrangements and unique chromosome features. The insertion and removal of transposons of Transposons are not imprecise which can indirectly affect surrounding sequences. Recombination between highly homologous regions dispersed by transposons at distant genomic positions lead to large-scale deletions, duplications, and inversions.
5. There is an intrinsic balance between transposon expression and repression. A variety of silencing mechanisms such as RNAi and epigenetics must be at least partially released to permit developmental regulation of host gene expression programs, particularly during early embryonic development.
6. Transposons are insertional mutagens in both germline and soma
7. Transposons can be damaging in ways that do not involve transposition. Reactivation of transposons interfere transcription and processing of host mRNAs. TE-encoded enzymes such as endonuclease may be lethal. Extrachromosomal DNA derived from transposons may trigger innate immune response.
8. A number of key coding and non-coding RNAs are derived from transposons. Transposon domestications.
9. Transposons contribute cis-regulatory DNA elements and modify transcriptional networks. 


#### 2. Types of transposons

![Imgur](https://imgur.com/5pq3fPx.jpg)
Piégu et al., 2015


![Imgur](https://imgur.com/BbY1tS1.jpg)

**Classes of mobile elements.** DNA transposons (e.g., Tc-1-mariner) have inverted terminal inverted repeats (ITRs) and a single open reading frame (ORF) that encodes a transposase. They are flanked by short direct repeats (DRs). Retrotransposons are divided into autonomous and nonautonomous classes depending on whether they have ORFs that encode proteins required for retrotransposition. Common autonomous retrotransposons are (i) LTRs or (ii) non-LTRs. Examples of LTR retrotransposons are human endogenous retroviruses (HERV) (shown) and various Ty elements of S. cerevisiae (not shown). These elements have terminal LTRs and slightly overlapping ORFs for their group-specific antigen (gag), protease (prt), polymerase (pol), and envelope (env) genes. They produce target site duplications (TSDs) upon insertion. Also shown are the reverse transcriptase (RT) and endonuclease (EN) domains. Other LTR retrotransposons that are responsible for most mobile-element insertions in mice are the intracisternal A-particles (IAPs), early transposons (Etns), and mammalian LTR-retrotransposons (MaLRs). These elements are not present in humans, and essentially all are defective, so the source of their RT in trans remains unknown. L1 is an example of a non-LTR retrotransposon. L1s consist of a 5'-untranslated region (5' UTR) containing an internal promoter, two ORFs, a 3' UTR, and a poly(A) signal followed by a poly(A) tail (An). L1s are usually flanked by 7- to 20-bp target site duplications (TSDs). The RT, EN, and a conserved cysteine-rich domain (C) are shown. An Alu element is an example of a nonautonomous retrotransposon. Alus contain two similar monomers, the left (L) and the right (R), and end in a poly(A) tail. Approximate full-length element sizes are given in parentheses.
© 2004 American Association for the Advancement of Science Kazasian, H. H. Mobile elements: drivers of genome evolution. Science 303, 1626–1632 (2004).