---
layout:     post
title:      生命科学系列
subtitle:   6-生物信息学实战
date:       2020-01-12
author:     Yingshan Li
header-img: img/sunrise.jpg
catalog: true
tags:
    - 生物信息学
---

### 1. Common formats of biological data

##### Fasta
The first line in a FASTA file started either with a ">" (greater-than) symbol. Following the initial line (used for a unique description of the sequence) was the actual sequence itself in standard one-letter character string. 

```
>gi|5524211|gb|AAD44166.1| cytochrome b [Elephas maximus maximus]
LCLYTHIGRNIYYGSYLYSETWNTGIMLLLITMATAFMGYVLPWGQMSFWGATVITNLFSAIPYIGTNLV
EWIWGGFSVDKATLNRFFAFHFILPFTMVALAGVHLTFLHETGSNNPLGLTSDSDKIPFHPYYTIKDFLG
LLILILLLLLLALLSPDMLGDPDNHMPADPLNTPLHIKPEWYFLFAYAILRSVPNKLGGVLALFLSIVIL
GLMPFLHTSKHRSMMLRPLSQALFWTLTMDLLTLTWIGSQPVEYPYTIIGQMASILYFSIILAFLPIAGX
IENY
```

##### Fastq

A FASTQ file normally uses four lines per sequence.
	* Line 1 begins with a '@' character and is followed by a sequence identifier and an optional description (like a FASTA title line).
	* Line 2 is the raw sequence letters.
	* Line 3 begins with a '+' character and is optionally followed by the same sequence identifier (and any description) again.
	* Line 4 encodes the quality values for the sequence in Line 2, and must contain the same number of symbols as letters in the sequence.
A FASTQ file containing a single sequence might look like this:

```
@SEQ_ID
GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT
+
!''*((((***+))%%%++)(%%%%).1***-+*''))**55CCF>>>>>>CCCCCCC65
```


##### SAM file

Biological sequences aligned to a reference sequence. 每个read占一行，共有12列：

```
1.read的名称
2.SAM标记
3.chromosome
4.5'端起始位置
5.MAPQ（mapping quality，比对的质量，数字越大，特异性越高）
6.CIGAR字串，记录插入、删除、错配以及splice junctions
7.mate名称，记录mate的信息
8.mate的位置
9.模版的长度
10.read序列
11.read质量
12.程序用标记
```
```
HWI-ST1001:137:C12FPACXX:7:1115:14131:66670     0       chr1    12805   1       42M4I5M *       0       0       TTGGATGCCCCTCCACACCCTCTTGATCTTCCCTGTGATGTCACCAATATG     CCCFFFFFHHGHHJJJJJHJJJJJJJJJJJJJJJJIJJJJJJJJJJJJIJJ     AS:i:-28        XN:i:0  XM:i:2  XO:i:1XG:i:4   NM:i:6  MD:Z:2C41C2     YT:Z:UU NH:i:3  CC:Z:chr15      CP:i:102518319  XS:A:+  HI:i:0
```

### 2. Common analysis protocol of high-throughput sequencing data

##### Obtaining data

从NCBI SRA下载测序数据或者实验室自己测的序，获得fastq文件

```
fastq-dump -X 5 -Z SRR390728
Prints the first five spots (-X 5) to standard out (-Z). This is a useful starting point for verifying other formatting options before dumping a whole file.

fastq-dump -I --split-files SRR390728
Produces two fastq files (--split-files) containing ".1" and ".2" read suffices (-I) for paired-end data.

fastq-dump --split-files --fasta 60 SRR390728
Produces two (--split-files) fasta files (--fasta) with 60 bases per line ("60" included after --fasta).
```

##### Quality report

```
fastqc seqfile1.fq
```


##### remove adaptors

```
fastx_clipper -Q 33 -l 18 -a ATCTCGTATGCCGTCTTCTGCTTG -i seqfile1.fq -o seqfile1_clipped.fq
```
	
More information about adaptors: https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary

Other command lines for fasts-toolkit:Convert FASTQ files to FASTA files. 

```
fastq_to_fasta -h
usage: fastq_to_fasta [-h] [-r] [-n] [-v] [-z] [-i INFILE] [-o OUTFILE]
```

##### Align the clipped sequences to the reference genome


**1. Build genome index**

```
bowtie2-build Reference_genome Reference_index
```

**2. Mapping**

```
bowtie2 -x Reference_index -U seqfile1_clipped.fq -S seqfile1_output.sam
bowtie2 -x index-name -1 leftreads.fa -2 rightreads.fa output.sam
```

### 3. biopieces

Some handy bioinformatics tools such as biopieces ((https://github.com/maasha/biopieces) can help us do it all in one line (https://github.com/maasha/biopieces): 

```
read_fastq -i RNA_virome.fa|find_adaptor -L 1 -f TCAGACGCTCGACA -r GGGGATAGG |clip_adaptor |trim_seq -m 30| grab -e 'SEQ_LEN > 20' mean_scores|write_fastq -o RNA_output.fa > RNA_output.txt
```


### 4. Local homology search (blast)
    
**1. Make local blast database:**

```
makeblastdb -in refseq_nucl.fa -dbtype nucl -parse_seqids -out refuseq_nucl
makeblastdb -in refseq_protein.fa -dbtype prot -parse_seqids -out refseq_protein
```

**2. Homology search:**

```
blastn -query query.fa -db refuseq_nucl -task megablast -num_threads 8 -evalue 1e-5 -parse_deflines > blastn.txt
blastn -query query.fa -db  refuseq_nucl -evalue 1e-5 -parse_deflines -outfmt "6 qseqid sseqid length qstart qend sstart send evalue bitscore" | sort -n -r -k 9 > blastn_summary.txt
```


### 5. Genome assembly

```
spades.py -m 16 --only-assembler -k 21,33,45 -1 seq_1.fq.gz -2 seq_2.fq.gz -o seq_contigs.fa
```


